### ISSUE TITLE
CA not consistent between library and GUI

### ISSUE THREAD

**Issue by JackDC89** [biolab/orange3#4139]
**Describe the bug**
The results (classification accuracy) of the analysis of a csv file using the GUI and the python library are not consistent. I am getting 0.550 CA with the GUI and 0.5 when invoking _Orange.classification.LogisticRegressionLearner_ from the python Library.

**To Reproduce**
Using the GUI, load file test1.xlsx and reproduce the simple program shown in Figure
![image](https://user-images.githubusercontent.com/7642636/67442302-826b1700-f600-11e9-820c-d84080c30468.png)
The test and score using a cross validation with 10 folds, no-stratified, will give 0.550 of CA

[test1.xlsx](https://github.com/biolab/orange3/files/3765018/test1.xlsx)
The same file, using a python program that calls _Orange.classification.LogisticRegressionLearner_ from the orange library, using a _KFold_ from _sklearn.model_selection_ will give a CA of 0.5.
Using the sklearn module for logistic regression also results in CA = 0.5 

**Orange version:**
3.23.1, but is was happening also with 3.20

**Expected behavior**
The same outcome from GUI and Python Library would be expected

**Screenshots**

**Operating system:**
Windows 10

**Additional context**
The difference in CA might seem negligible, but this is just simple example using a subset of features from the _heart_disease.tab_ dataset. The difference on my real dataset is much more pronounced.
I also attach a stand alone python script to show the procedure I used.
[test_standalone.zip](https://github.com/biolab/orange3/files/3765056/test_standalone.zip)
This is my first post, so I apologize for any formatting/etiquette error

**Comment by ajdapretnar** [comment]
I suspect the reason for this is that we shuffle the data in `Orange.evaluation.testing.py`, L568. Thus the splits are not the same. Default for `skl.StratifiedKFold` is shuffle=False. Probably we should set it to False as well. But I am no expert here. @janezd @markotoplak @lanzagar

**Comment by markotoplak** [comment]
@ajdapretnar, in Orange we mostly assume that samples are independent and therefore we should shuffle.

But it depends on the specific problem. See https://scikit-learn.org/stable/modules/cross_validation.html#a-note-on-shuffling

**Comment by ajdapretnar** [comment]
Thanks for the link!

I am still confused why it seems that shuffling does not consider random seed. Why the two shuffles different (the one in the code and the one in Orange)?
